futaba/cogs/pingable/core.py

```diff
@@ -0,0 +1,94 @@
+#
+# cogs/pingable/core.py
+#
+# futaba - A Discord Mod bot for the Programming server
+# Copyright (c) 2017-2020 Jake Richardson, Ammon Smith, jackylam5
+#
+# futaba is available free of charge under the terms of the MIT
+# License. You are free to redistribute and/or modify it under those
+# terms. It is distributed in the hopes that it will be useful, but
+# WITHOUT ANY WARRANTY. See the LICENSE file for more details.
+#
+
+"""
+Cog for pingable helper roles
+"""
+
+import logging
+from datetime import datetime, timedelta
+
+import discord
+from discord.ext import commands
+from futaba.permissions import mod_perm
+from futaba.exceptions import CommandFailed
+from futaba.utils import fancy_timedelta
+from ..abc import AbstractCog
+
+logger = logging.getLogger(__name__)
+
+__all__ = ["Pingable"]
+
+
+class Pingable(AbstractCog):
+    __slots__ = ("journal",)
+
+    def __init__(self, bot):
+        super().__init__(bot)
+        self.journal = bot.get_broadcaster("/pingable")
+        self.cooldowns = {}
+
+    def setup(self):
+        pass
+
+    @commands.command(name="pinghelpers", aliases=["helpme", "pinghelp"])
+    async def pinghelpers(self, ctx):
+        """Pings helpers if used in the respective channel"""
+
+        cooldown_time = self.bot.config.helper_ping_cooldown
+
+        logger.info(
+            "User '%s' (%d) is pinging the helper role in channel '%s' in guild '%s' (%d)",
+            ctx.author,
+            ctx.author.mention,
+            ctx.channel,
+            ctx.guild,
+            ctx.guild.id,
+        )
+        pingable_channels = self.bot.sql.roles.get_pingable_role_channels(ctx.guild)
+        # this will return an empty list if there is nothing.
+        channel_role = [
+            (channel, role)
+            for channel, role in pingable_channels
+            if channel == ctx.channel
+        ]
+
+        if not channel_role:
+            embed = discord.Embed(colour=discord.Colour.red())
+            embed.set_author(name="Failed to ping helper role.")
+            embed.description = str(f"There is no helper role set for this channel.")
+            raise CommandFailed(embed=embed)
+
+        channel_user = (ctx.channel.id, ctx.author.id)
+        cooldown = self.cooldowns.get(channel_user)
+
+        if mod_perm(ctx) or not cooldown or cooldown <= datetime.now():
+            self.cooldowns[channel_user] = datetime.now() + timedelta(
+                seconds=cooldown_time
+            )
```

Comment:
are you sure you want to use `datetime` instead of just storing `time.time()` here?

Reply:
`datetime` is used elsewhere in the codebase so using a different type here seems arbitrary. also iirc pylint complains about `time.time()`